ddtrace/tracer.py

```diff
@@ -411,10 +411,12 @@ def start_span(
             span = tracer.start_span('web.worker', child_of=context)
         """
         new_ctx = self._check_new_process()
+        if new_ctx:
+            child_of = new_ctx
 
         if child_of is not None:
             if isinstance(child_of, Context):
-                context = new_ctx or child_of
+                context = child_of
                 parent = child_of._get_current_span()
             else:
                 context = child_of.context
```

Comment:
do we need to do the same here?

would it be better to have:

```python
context = child_of if isinstance(child_of, Context) else child_of.context
parent = child_of._get_current_span()
```

?

although, I suppose it doesn't matter to grab the current span if we have it?

Reply:
Not sure I follow. In that `else` branch `child_of` is a span 🤔